core/config/toml/types.go

```diff
@@ -1301,3 +1304,82 @@ func (t *ThresholdKeyShareSecrets) validateMerge(f *ThresholdKeyShareSecrets) (e
 
 	return err
 }
+
+type Tracing struct {
+	Enabled         *bool
+	CollectorTarget *string
+	NodeID          *string
+	SamplingRatio   *float64
+	Attributes      *map[string]string
```

Comment:
Hmm, good question. I lean towards validating the config even if tracing is not enabled. What do you think?

Reply:
I mean that I think you are getting default zero values filled automatically, in which case it would be disabled, so limiting validation to enabled only would avoid the problem.

Reply:
> I lean towards validating the config even if tracing is not enabled. What do you think?

I think that when we see 'not enabled', it could just have been omitted entirely by the user.

Reply:
btw - setting Attributes to the non-pointer type breaks assumptions around non-nil fields for TestConfig_setDefaults in core/services/chainlink/config_test. 

In general, our TOML/config testing is very brittle and we have low cohesion here. Definitely something I want to continue looking into in the future

Reply:
This is the first case of map so it was not considered previously. We may have to add support for it, but I don't think we can necessarily just defer on that. We need to at least understand what is happening before committing this to develop.

Reply:
As a workaround, can we at least control the logged output by niling out the map when empty, or something like that? From `TOMLString`? https://github.com/smartcontractkit/chainlink/blob/697150e43fa847f926d641b9a6ab86577b45547f/core/services/chainlink/config.go#L47

Reply:
For now I've just added 
`c.Tracing.Attributes = make(map[string]string)` to the test. I think it's cleaner to have hacks in the testing logic?

Reply:
But the point is that the output is producing an unsatisfactory user experience. We need hacks in production until/unless we can add map support to the other logic.

Reply:
Wait, do you mean that a nil map (non-pointer) prints the header, but a zero value map does not print anything?!
That sounds like a bug in go-toml.

Reply:
> Wait, do you mean that a nil map (non-pointer) prints the header, but a zero value map does not print anything?!

Yep, that's at least what I'm observing right now. 

> But the point is that the output is producing an unsatisfactory user experience

The non-pointer type should take care of the hanging header.

Reply:
> The non-pointer type should take care of the hanging header.

Ah, OK, I misunderstood. TOML looks clean now :+1: